app/controllers/books_controller.rb

```diff
@@ -46,8 +46,8 @@ def create
   # PATCH/PUT /books/1
   # PATCH/PUT /books/1.json
   def update
-    @book.cover_image.attach(params[:cover_image])
-    @book.content.attach(params[:content])
+    @book.cover_image.attach(params[:cover_image]) if !(params[:cover_image].nil?)
```

Comment:
Not sure if this makes sense. Should we be attaching `@book.cover_image.attach(params[:cover_image]) if params[:cover_image].present?`

Because if `params[:cover_image]` is `nil`, it means that user didn't send a cover image via the form

Reply:
Yep. Covers the case where when updating book no content or cover image is supplied (i.e. use the existing data).

Reply:
can be `@book.cover_image.attach(params[:cover_image]) unless params[:cover_image].nil?`. The current implementation is a double negative